src/System Application/App/File System/src/Account/FileAccountWizard.Page.al

```diff
@@ -0,0 +1,493 @@
+        AppSourceAvailable := AppSource.IsAvailable();
+        LoadTopBanners();
+    end;
+
+    local procedure NextStep(Backwards: Boolean)
+    begin
+        if Backwards then
+            Step -= 1
+        else
+            Step += 1;
+
+        SetDefaultControls();
+
+        case Step of
+            Step::Welcome:
+                ShowWelcomeStep();
+            Step::"Choose Connector":
+                ShowChooseConnectorStep();
+            Step::"Register Account":
+                ShowRegisterAccountStep();
+            Step::Done:
+                ShowDoneStep();
+        end;
+    end;
+
+    local procedure ShowWelcomeStep()
+    begin
+        WelcomeVisible := true;
+        BackActionEnabled := false;
+    end;
+
+    local procedure ShowChooseConnectorStep()
+    begin
+        if not ConnectorsAvailable then
+            NextActionEnabled := false;
+
+        ChooseConnectorVisible := true;
+    end;
+
+    local procedure ShowRegisterAccountStep()
+    var
+        FeatureTelemetry: Codeunit "Feature Telemetry";
+        Telemetry: Codeunit Telemetry;
+        AccountWasRegistered: Boolean;
+        ConnectorSucceeded: Boolean;
+        CustomDimensions: Dictionary of [Text, Text];
+        CTHLbl: Label '%1 account has been setup.', Locked = true;
+        CTILbl: Label '%1 account has failed to setup. Error: %2', Locked = true;
+    begin
+        ConnectorSucceeded := TryRegisterAccount(AccountWasRegistered);
+        CustomDimensions.Add('Category', FileCategoryLbl);
+
+        if AccountWasRegistered then begin
+            FeatureTelemetry.LogUptake('0000CTF', 'File Access', Enum::"Feature Uptake Status"::"Set up");
+            Telemetry.LogMessage('0000CTH', StrSubstNo(CTHLbl, Rec.Connector), Verbosity::Normal, DataClassification::SystemMetadata, TelemetryScope::ExtensionPublisher, CustomDimensions);
+            NextStep(false);
+        end else begin
+            Telemetry.LogMessage('0000CTI', StrSubstNo(CTILbl, Rec.Connector, GetLastErrorCallStack()), Verbosity::Normal, DataClassification::SystemMetadata, TelemetryScope::ExtensionPublisher, CustomDimensions);
+            NextStep(true);
+        end;
+
+        if not ConnectorSucceeded then
+            Error(GetLastErrorText());
+    end;
+
+    [TryFunction]
+    local procedure TryRegisterAccount(var AccountWasRegistered: Boolean)
+    var
+        FileAccountImpl: Codeunit "File Account Impl.";
+        FileConnector: Interface "File System Connector";
+    begin
+        // Check to validate that the connector is still installed
+        // The connector could have been uninstalled by another user/session
+        if not FileAccountImpl.IsValidConnector(Rec.Connector) then
+            Error(FileConnectorHasBeenUninstalledMsg);
+
+        FileConnector := Rec.Connector;
+
+        ClearLastError();
+        AccountWasRegistered := FileConnector.RegisterAccount(RegisteredAccount);
```

Comment:
Doesnt this result in a Database write operation? Shouldnt this be avoided within tryfunctions?
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-handling-errors-using-try-methods#DbWriteTransactions

Reply:
You are right. But for some reason it doesn't. It is the same in the email module:
https://github.com/microsoft/BCApps/blob/4c7b6487cb55b1b750997f3ef33b1a92f8e678b8/src/System%20Application/App/Email/src/Account/EmailAccountWizard.Page.al#L510

And booth wizards work well in my onprem enviroment, there transactions are not allowed in try functions ...

Reply:
It seems to work because the Registered Account is Temporary. IMHO it would still make sense to mention this in the documentation on the interface for that procedure, so it does not get overlooked

`/// <remarks>The out parameter must hold the account ID of the added account. This procedure is called from within a TryFunction, any database transaction IMD will result in a runtime error.</remarks>`